### PR TITLE
Upgrade Elasticsearch client to 8.6.1

### PR DESCRIPTION
**Pull request by b-deam**
Reverting the revert, also includes a keyword argument fix for the missed `get_transform_stats` operation in Telemetry.

Relates: https://github.com/elastic/rally/pull/1669

### CODE REVIEW

File: esrally/driver/runner.py [hunk 1]
```diff
@@ -2142,13 +2081,13 @@ class RestoreSnapshot(Runner):
...
-        await es.snapshot.restore(
-            repository=mandatory(params, "repository", repr(self)),
-            snapshot=mandatory(params, "snapshot", repr(self)),
-            wait_for_completion=params.get("wait-for-completion", False),
-            **api_kwargs,
-        )
+        repo = mandatory(params, "repository", repr(self))
+        snapshot = mandatory(params, "snapshot", repr(self))
+
+        await es.perform_request(method="POST", path=f"/_snapshot/{repo}/{snapshot}/_restore", **api_kwargs)
```
Comment (dliappis) [inline esrally/driver/runner.py:2090]: Should we leave a comment about why we aren't using `es.snapshot.restore` along with a TODO to fix in the future as soon as the client supports the missing parameter?

**Review by dliappis — COMMENTED** [review]
Thanks for following up. Left a small ask so that we can follow up/clean up in the future.

File: esrally/driver/runner.py (hunk 1 above)
Comment (b-deam) [reply esrally/driver/runner.py:2090]: Yes! I forgot to add it and pushed so I could test some esbench use-cases.

**Review by pquentin — APPROVED** [review]
:fire:

**Pull request merged**